sqldelight-idea-plugin/src/main/kotlin/com/squareup/sqldelight/intellij/actions/GenerateSelectAllQueryAction.kt

```diff
@@ -0,0 +1,50 @@
+package com.squareup.sqldelight.intellij.actions
+
+import com.alecstrong.sql.psi.core.psi.SqlCreateTableStmt
+import com.intellij.codeInsight.CodeInsightActionHandler
+import com.intellij.codeInsight.actions.CodeInsightAction
+import com.intellij.codeInsight.template.TemplateManager
+import com.intellij.codeInsight.template.impl.TemplateManagerImpl
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.editor.ScrollType
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiFile
+import com.intellij.psi.PsiWhiteSpace
+import com.intellij.psi.util.PsiTreeUtil
+import com.intellij.psi.util.parentOfType
+import com.squareup.sqldelight.core.lang.SqlDelightFile
+import org.jetbrains.kotlin.psi.psiUtil.endOffset
+
+class GenerateSelectAllQueryAction : CodeInsightAction() {
+
+  override fun isValidForFile(project: Project, editor: Editor, file: PsiFile): Boolean {
+    val offset = editor.caretModel.offset
+    return file is SqlDelightFile && file.findElementAt(offset)?.parentOfType<SqlCreateTableStmt>() != null
+  }
+
+  override fun getHandler(): CodeInsightActionHandler = SelectAllHandler()
+
+  class SelectAllHandler : CodeInsightActionHandler {
+    override fun invoke(project: Project, editor: Editor, file: PsiFile) {
+      val caretOffset = editor.caretModel.offset
+      val createTableStmt = file.findElementAt(caretOffset)?.parentOfType<SqlCreateTableStmt>() ?: return
+
+      val tableName = createTableStmt.tableName.name
+      val selectAllTemplate = TemplateManagerImpl.listApplicableTemplates(file, caretOffset, false)
+        .first { it.key == "sel" }
+
+      val document = editor.document
+      var lastChild = file.lastChild
+      if (lastChild is PsiWhiteSpace) {
+        lastChild = PsiTreeUtil.skipWhitespacesAndCommentsBackward(lastChild)
+      }
+      document.insertString(lastChild.endOffset, "\n\n")
```

Comment:
it might be nice to also auto-gen the identifier for it, like

```
selectAll:
SELECT * FROM test;
```

Reply:
Done

Reply:
hm maybe we should place a cursor on it?

Reply:
i think ideal scenario is it puts `selectAll:` for you but the `selectAll` part is highlighted so you can type your own thing if you want.

Reply:
yeah, I will make that

Reply:
https://user-images.githubusercontent.com/14963604/116009299-701ccf80-a621-11eb-95a6-f1a6c755d1ed.mov

Reply:
This is amazing.

When I do an operation like "extract variable" the new variable is both selected but also somehow marked as like an editable region or something. I don't know what advantages it offers over simple selection...

https://user-images.githubusercontent.com/66577/116017049-88362400-a60c-11eb-8d55-ee86a5c05ea9.mov

Are we able to do this?

Reply:
the shortest way would be to modify the "sel" live template since I'm reusing it here. is that ok?

Reply:
https://user-images.githubusercontent.com/14963604/116031232-e8a57f80-a665-11eb-8ea7-a55d81e64013.mov

Reply:
i dig it

Reply:
done